### PR TITLE
Call hierarchy incoming calls include also incoming calls of super methods

### PR DESCRIPTION
**Pull request by kasiaMarek**
Resolves https://github.com/scalameta/metals/issues/4488

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/callHierarchy/CallHierarchyProvider.scala [hunk 1]
```diff
@@ -103,8 +107,16 @@ final case class CallHierarchyProvider(
       params: CallHierarchyIncomingCallsParams,
       token: CancelToken,
   )(implicit ec: ExecutionContext): Future[List[CallHierarchyIncomingCall]] = {
-    val info = getInfo(params.getItem.getData)
-
+    val parents = supermethods
+      .getSuperMethodHierarchySymbols(
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/callHierarchy/CallHierarchyProvider.scala:111]: Maybe we could create an additiona `getSuperMethodHierarchySymbols` method that would take in AbsolutePath and Position? We wouldn't need to create TextDocumentPositionParams

**Review by tgodzik — CHANGES_REQUESTED** [review]
Tiny nitpick, but otherwise great work!

File: metals/src/main/scala/scala/meta/internal/metals/callHierarchy/Demo.scala [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package scala.meta.internal.metals.callHierarchy
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/callHierarchy/Demo.scala:1]: Looks like this was added by mistake during tests? Could you remove it?

Comment (kasiaMarek) [reply metals/src/main/scala/scala/meta/internal/metals/callHierarchy/Demo.scala]: agrh...

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**